src/column_store/ColumnStoreTable_AVX.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <algorithm>
+#include <vector>
+
+#include "ITable_AVX.h"
+#include "ColumnStoreHelper.h"
+
+// whether to check the remainder in the filterBasic()-method
+// this could have an impact on performance due to unfortunate branch prediction
+// TODO benchmark performance
+#define WITH_R_0_CHECK 1
+
+namespace ColumnStore::AVX {
+    template<typename T>
+    class Table : public Tables::AVX::ITable<T> {
+    public:
+        Table(uint64_t numAttributes, uint64_t numRows, const T **initialData)
+                : Tables::AVX::ITable<T>(numAttributes, numRows, initialData) {
+            data = new std::vector<T>[numAttributes];
```

Comment:
I guess it would be possible to share quite some code between the AVX version and the scalar version. All methods that are not directly used for filtering should be the same, right? 

While column store is probably a little bit easier, I don't feel good about duplicating a lot of the PAX code for the scalar and the AVX version.

I would propose to have a `BaseColumnStore` the contains all non-filtering and non-projecting methods and inherit two classes, a `ScalarColumnStore` and an `AvxColumnStore`.

Reply:
This sounds nice on paper, however it would require the `Filters::Scalar` filters and `Filters::AVX` filters to be combined. Currently there are two `ITable.h`: one using AVX filters and one using the scalar implementation - this means, that I cannot make a base class for column store since I dont know what `ITable` I should inherit from at this point.

This means that we either need to combine the filters ore live with copied code. Maybe some of the code can be transported into a separate header with helper methods? This header could then be used by both implementations.

Reply:
I don't think it would require a combined filter 🧐 But I guess we can think about this later, I don't want to block this PR. Later, I open a PR with what I was trying to say..